### PR TITLE
PKG-2688 rebuild for py311

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
We have to get back to 1.12.0 and rebuild it for python 3.11 as this is dependency for snowflake-telemetry-python
- create new main-1.12.0 branch
- bump up build number
- update python version

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
Tests failed for `TEST START: /upload/linux-aarch64/opentelemetry-exporter-otlp-proto-grpc-1.12.0-py311hd43f75c_3.tar.bz2` because of a dependency conflict for python 3.11

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Review by boldorider4 — COMMENTED** [review]
There is a series of linter errors that need to be fixed

File: recipe/meta.yaml [hunk 1]
```diff
@@ -88,8 +88,8 @@ source:
     - two-patches.patch
 
 build:
-  number: 2
-  skip: true # [py<36]
+  number: 3
+  skip: true # [py<37]
```
Comment (boldorider4) [inline recipe/meta.yaml:92]: why `py<37`? Is this coming from upstream?

**Comment by M-Waszkiewicz-Anaconda** [comment]
> There is a series of linter errors that need to be fixed

the scope of the ticket is just about rebuild for py 311, It is so much complicated I do not want to touch it as was due in previous sprint

File: recipe/meta.yaml (hunk 1 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:92]: cannot find anything about python requirements, but probably we don't want to support so ancient version and/or spoil snowflake channel

**Review by cbouss — APPROVED** [review]
This looks alright.

Two details, which don't change the end result in that case. (i.e. no obligation to make the change)
`- dataclasses ==0.8  # [py<3.7]`
--> `- dataclasses ==0.8  # [py<37]`

` pip install . -vv --no-deps`
-->  `pip install . -vv --no-deps --no-build-isolation`

File: recipe/meta.yaml (hunk 1 above)
Comment (boldorider4) [reply recipe/meta.yaml:92]: so why not remove it altogether since we build `py>=3.8`? Unless the upstream is saying something specific

**Pull request merged**